activerecord/lib/active_record/associations/collection_proxy.rb

```diff
@@ -1149,6 +1148,18 @@ def find_from_target?
         def exec_queries
           load_target
         end
+
+        def respond_to_missing?(method, _)
+          scope.respond_to?(method) || super
```

Comment:
@rafaelfranca no; our `method_missing` checks `respond_to?` (without allowing privates) and then uses `public_send`: we don't respond to `scope`'s private methods, even when invoked privately.